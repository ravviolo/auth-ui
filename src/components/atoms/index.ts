export { Button } from './Button/Button';
export { Checkbox } from './Checkbox/Checkbox';
export { Divider } from './Divider/Divider';
export { Heading } from './Heading/Heading';
export { IconButton } from './IconButton/IconButton';
export { InputField } from './InputField/InputField';
export { InputLabel } from './InputLabel/InputLabel';
export { TextButton } from './TextButton/TextButton';
