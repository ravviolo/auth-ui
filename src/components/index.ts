export * from './templates/LoginTemplate/LoginTemplate';
export * from './templates/SignUpTemplate/SignUpTemplate';
