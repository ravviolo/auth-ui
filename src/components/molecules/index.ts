export { AuthFooter } from './AuthFooter/AuthFooter';
export { AuthHeader } from './AuthHeader/AuthHeader';
export { AuthSocials } from './AuthSocials/AuthSocials';
export { EmailInput } from './EmailInput/EmailInput';
export { PasswordInput } from './PasswordInput/PasswordInput';
