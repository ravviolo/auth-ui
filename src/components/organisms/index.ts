export * from './LoginForm/LoginForm';
export * from './SignUpForm/SignUpForm';
