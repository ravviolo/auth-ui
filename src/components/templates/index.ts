export * from './LoginTemplate/LoginTemplate';
export * from './SignUpTemplate/SignUpTemplate';
