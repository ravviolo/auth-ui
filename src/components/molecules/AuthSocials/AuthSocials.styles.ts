import styled from 'styled-components';

import { Flex, IconButton } from 'components/atoms';

export const Container = styled(Flex)`
  align-items: center;
  justify-content: center;
  gap: 1rem;
`;

export const AuthSocialButton = styled(IconButton)<{ color: string }>`
  color: ${({ color }) => color};
  border-color: ${({ color }) => color};
`;
