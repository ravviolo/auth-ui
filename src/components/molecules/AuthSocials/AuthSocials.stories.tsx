import { ComponentMeta, ComponentStory } from '@storybook/react';

import { AuthSocials } from 'components/molecules';

export default {
  title: 'Molecules/AuthSocials',
  component: AuthSocials,
} as ComponentMeta<typeof AuthSocials>;

export const Default: ComponentStory<typeof AuthSocials> = (args) => (
  <AuthSocials {...args} />
);
Default.storyName = 'AuthSocials';
