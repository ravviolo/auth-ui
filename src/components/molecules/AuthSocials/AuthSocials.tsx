import { IconButton } from 'components/atoms';

import * as S from './AuthSocials.styles';

interface Props {
  onClickGoogle: () => void;
  onClickFacebook: () => void;
  onClickLinkedIn: () => void;
}

interface AuthSocial {
  name: string;
  color: string;
  icon: string;
  onClickHandler: () => void;
}

export const AuthSocials = ({
  onClickFacebook,
  onClickGoogle,
  onClickLinkedIn,
}: Props) => {
  const authSocials: AuthSocial[] = [
    {
      name: 'Google',
      color: '#DB4437',
      icon: 'G',
      onClickHandler: onClickGoogle,
    },
    {
      name: 'Facebook',
      color: '#4267B2',
      icon: 'f',
      onClickHandler: onClickFacebook,
    },
    {
      name: 'LinkedIn',
      color: '#0077B5',
      icon: 'in',
      onClickHandler: onClickLinkedIn,
    },
  ];

  return (
    <S.Container direction="row">
      {authSocials.map(({ color, icon, name, onClickHandler }) => (
        <IconButton
          key={name}
          color={color}
          icon={icon}
          type="button"
          onClick={onClickHandler}
        />
      ))}
    </S.Container>
  );
};
