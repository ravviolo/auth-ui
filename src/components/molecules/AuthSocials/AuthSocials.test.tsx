import { render, screen, within } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { create } from 'react-test-renderer';

import { AuthSocials } from 'components/molecules';

import { Props as AuthSocialsProps } from './AuthSocials';

describe('AuthSocials', () => {
  const mockOnClickGoogle = jest.fn<AuthSocialsProps['onClickGoogle'], []>();
  const mockOnClickFacebook = jest.fn<
    AuthSocialsProps['onClickFacebook'],
    []
  >();
  const mockOnClickLinkedIn = jest.fn<
    AuthSocialsProps['onClickLinkedIn'],
    []
  >();

  const props: AuthSocialsProps = {
    testId: 'auth-socials-test-id',
    onClickFacebook: mockOnClickFacebook,
    onClickGoogle: mockOnClickGoogle,
    onClickLinkedIn: mockOnClickLinkedIn,
  };

  it('should render icon buttons for authenticating through socials', () => {
    render(<AuthSocials {...props} />);

    const authSocials = screen.getByTestId('auth-socials-test-id');

    const socialsIconBtns = within(authSocials).getAllByTestId(
      /icon-btn-([A-Z])\w+-test-id/i
    );

    expect(socialsIconBtns).toHaveLength(3);
  });

  it('should handle click events on Facebook icon - run handler once, not trigger other handlers', async () => {
    render(<AuthSocials {...props} />);

    const authSocials = screen.getByTestId('auth-socials-test-id');
    const facebookIconBtn = within(authSocials).getByTestId(
      /icon-btn-facebook-test-id/i
    );

    await userEvent.click(facebookIconBtn);

    expect(mockOnClickFacebook).toHaveBeenCalledTimes(1);
    expect(mockOnClickGoogle).not.toBeCalled();
    expect(mockOnClickLinkedIn).not.toBeCalled();
  });

  it('should handle click events on Google icon - run handler once, not trigger other handlers', async () => {
    render(<AuthSocials {...props} />);

    const authSocials = screen.getByTestId('auth-socials-test-id');
    const googleIconBtn = within(authSocials).getByTestId(
      /icon-btn-google-test-id/i
    );

    await userEvent.click(googleIconBtn);

    expect(mockOnClickGoogle).toHaveBeenCalledTimes(1);
    expect(mockOnClickFacebook).not.toBeCalled();
    expect(mockOnClickLinkedIn).not.toBeCalled();
  });

  it('should handle click events on LinkedIn icon - run handler once, not trigger other handlers', async () => {
    render(<AuthSocials {...props} />);

    const authSocials = screen.getByTestId('auth-socials-test-id');
    const linkedInIconBtn = within(authSocials).getByTestId(
      /icon-btn-linkedin-test-id/i
    );

    await userEvent.click(linkedInIconBtn);

    expect(mockOnClickLinkedIn).toHaveBeenCalledTimes(1);
    expect(mockOnClickFacebook).not.toBeCalled();
    expect(mockOnClickGoogle).not.toBeCalled();
  });

  it('should match snapshot', () => {
    const tree = create(<AuthSocials {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
