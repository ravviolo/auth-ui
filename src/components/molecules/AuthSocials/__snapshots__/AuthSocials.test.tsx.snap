// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthSocials should match snapshot 1`] = `
<div
  className="sc-jSMfEi iChCMC sc-evZas fBPmfS"
  data-testid="test-auth-socials"
  direction="row"
>
  <button
    className="sc-iBkjds cxbDaZ"
    color="#DB4437"
    data-testid="icon-btn-Google"
    onClick={[MockFunction]}
    type="button"
  >
    G
  </button>
  <button
    className="sc-iBkjds bqhZzF"
    color="#4267B2"
    data-testid="icon-btn-Facebook"
    onClick={[MockFunction]}
    type="button"
  >
    f
  </button>
  <button
    className="sc-iBkjds hAqsSt"
    color="#0077B5"
    data-testid="icon-btn-LinkedIn"
    onClick={[MockFunction]}
    type="button"
  >
    in
  </button>
</div>
`;
